code/game/machinery/constructable_frame.dm

```diff
@@ -153,7 +153,7 @@
 			if(istype(P, /obj/item))
 				var/success
 				for(var/I in req_components)
-					if(istype(P, I) && (req_components[I] > 0))
+					if(istype(P, I) && (req_components[I] > 0) && !(P.flags & NODROP))

```

Comment:
Honestly, I'd just stick with this. It's much more readable and it doesn't matter much.